java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/core/DbzSourceHandler.java

```diff
@@ -46,39 +46,57 @@ public void startSource(ServerCallStreamObserver<GetEventStreamResponse> respons
             // Start the engine
             runner.start();
             LOG.info("Start consuming events of table {}", config.getSourceId());
-            while (runner.isRunning()) {
-                try {
-                    // Thread will block on the channel to get output from engine
-                    var resp =
-                            runner.getEngine().getOutputChannel().poll(500, TimeUnit.MILLISECONDS);
 
-                    if (resp != null) {
+            class OnReadyHandler implements Runnable {
```

Comment:
May move the class definition out of the function to make the code easy to read.